src/TodoList.js

```diff
@@ -1,37 +1,63 @@
 class TodoList {
-  constructor () {
+  constructor() {
     this.id = 0
     this.items = []
   }
 
-  create (str) {
+  create(str) {
     this.id++
-    const item = { id: this.id, text: str, status: 'incomplete' }
+    const item = {
+      id: this.id,
+      text: str,
+      status: 'incomplete',
+      created: new Date()
+    }
     this.items.push(item)
     return item
   }
 
-  showAll () {
-    return this.items
+  formatTextOverflow() {
+    const reducedText = [...this.items]
+
+    reducedText.forEach((item) => {
+      if (item.text.length >= 20) {
+        item.text = item.text.slice(0, 17) + '...'
+      }
+    })
+
+    return reducedText
```

Comment:
The `.map` array function returns a copy of the original array with the data modified so it doesn't mutate the original array, it's safe to use instead of copying the array manually and doing a forEach on the copied version